internal/accrual/client.go

```diff
@@ -0,0 +1,161 @@
+package accrual
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/ktigay/loyalty/internal/entity"
+)
+
+const (
+	orderInfoPath = "/api/orders/"
+	rateLimit     = 3
+	timeout       = 1 * time.Second
+)
+
+// Client Структура для работы с сервисом Accrual.
+type Client struct {
+	endpoint string
+	logger   *slog.Logger
+}
+
+// OrderStatus Данные по заказу из сервиса Accrual.
+func (s *Client) OrderStatus(ctx context.Context, orderID string) (*entity.Accrual, error) {
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	client := resty.New()
+	url := s.endpoint + orderInfoPath + orderID
+
+	e := entity.Accrual{}
+	resp, err := client.R().
+		SetContext(ctx).
+		SetResult(&e).
+		Get(url)
+	if err != nil {
+		return nil, RequestError{
+			StatusCode: 0,
+			URL:        url,
+			Message:    err.Error(),
+		}
+	}
+
+	code := resp.StatusCode()
+	if code == http.StatusOK {
+		return &e, nil
+	}
+
+	return nil, RequestError{
+		StatusCode: code,
+		URL:        url,
+	}
+}
+
+// OrdersStatus Данные по заказам из сервиса Accrual.
+func (s *Client) OrdersStatus(ctx context.Context, orderIDs []string) (*[]entity.Accrual, error) {
```

Comment:
Нет смысла возвращать указатель на массив так как массив сам по себе ссылочный тип и передается по ссылке.
```suggestion
func (s *Client) OrdersStatus(ctx context.Context, orderIDs []string) ([]entity.Accrual, error) {
```